### PR TITLE
Migrate to AndroidX; Bump gralde version

### PR DESCRIPTION
**Pull request by slavkoder**
- [x] Moved from old support library to AndroidX;
- [x] Upgraded Gradle plugin (helps avoid annoying/confusing dialog when you import a project)
- [x] Updated [accompanying Guide](https://github.com/codepath/android_guides/wiki/Using-the-App-Toolbar/_compare/509e6f4b3f2f3d9d13bae4d6a084eea1107f7e62...0224cf51baf41e378a3cac9083260b622f502fe4)
- [x] Updated [Extended Guide](https://github.com/codepath/android_guides/wiki/Extended-ActionBar-Guide/_compare/6bbc4163176f5dc28de6ff0abbea0260d5320515...fb187545088854698dd1faaa657fc6160f67053b)
- [x] Updated [Constructing View Layouts guide](https://github.com/codepath/android_guides/wiki/Constructing-View-Layouts/_compare/1c74517a6cbbe221f1f809a7d305d8633b5d2ad4...ca662ae7833067a39747f66a39d7d5c38d9fcbc3)
- [x] Update [Design Support Library guide](https://github.com/codepath/android_guides/wiki/Design-Support-Library/_compare/86ca4d835048b09bf26d8b371bfe545e22b8bd60...512467980f56f2a934bcadda4ab5690b04dfa9df) ⚠️  I deleted a bunch of parts there, let me know if it's okay.

### TODO
- [x] Migrate [FloatingActionButton](https://github.com/codepath/android_guides/wiki/Floating-Action-Buttons/_compare/509e6f4b3f2f3d9d13bae4d6a084eea1107f7e62...3b09582abc93a68259b0191ff0125a8b5c7fb390)
- [x] Migrate [RecyclerView guide](https://github.com/codepath/android_guides/wiki/Using-the-RecyclerView/_compare/5f4e445e18527d9826fd56c3a243b83a5d60e2c0...ca2eeb23aabc5e80e99fe4c3f194545306d72476)
- [x] All the Features that are referenced in Design Support Lib: https://guides.codepath.org/android/Design-Support-Library#features

### Or not TODO?
- [x] https://github.com/codepath/android-intents-toolbar-sample-app - Should I migrate/update this app?

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -19,10 +19,12 @@ android {
 }
 
 dependencies {
-    implementation 'com.android.support:appcompat-v7:28.0.0'
+    implementation 'androidx.appcompat:appcompat:1.0.0'
```
Comment (rogerhu) [inline app/build.gradle:22]: Is this the latest?  I thought it was 1.0.2.

**Comment by rogerhu** [comment]
We are also swapping out the async http client with an OK http based version.  Can you take a look at https://github.com/codepath/AsyncHttpClient?

Grade import will be 'com.codepath.libraries:asynchttpclient:0.0.5' though I will be making incremental improvements so the version may change

File: app/build.gradle (hunk 1 above)
Comment (slavkoder) [reply app/build.gradle:22]: that latest is `1.1.0` right now, but the [Migration guide](https://developer.android.com/jetpack/androidx/migrate) still points to `1.0.0`. Different dependencies have different versions now, so I use migration guide as source of truth for all of them.

**Comment by slavkoder** [comment]
@rogerhu do you think it makes sense to make version variables like this:
```
ext {
    // versions for dependencies
    asyncHttpVersion = '0.0.5'
    appCompatVersion = '1.0.0'
    ...
}
```

**Comment by rogerhu** [comment]
If you do make sure to put it close to the dependency declaration... to
minimize student confusion.

**Comment by nesquena** [comment]
The TODO looks good, also as you mentioned assuming it's not too time consuming I think it'd be good to upgrade https://github.com/codepath/android-intents-toolbar-sample-app and any other sample code linked from the course system to latest Gradle/Android/AndroidX

**Pull request merged**